packages/firestore/src/api/user_data_writer.ts

```diff
@@ -54,14 +54,23 @@ export type ServerTimestampBehavior = 'estimate' | 'previous' | 'none';
  * to the user.
  */
 export class UserDataWriter {
+  /**
+   * @param databaseId DatabaseID for this instance
+   * @param referenceFactory A factory function that returns a firestore-exp or
+   * classic DocumentReference
+   * @param bytesFactory A factory function that returns a firestore-exp Bytes
+   * or a classic Blob type
+   */
   constructor(
     private readonly databaseId: DatabaseId,
-    private readonly serverTimestampBehavior: ServerTimestampBehavior,
     private readonly referenceFactory: (key: DocumentKey) => unknown,
     private readonly bytesFactory: (bytes: ByteString) => Bytes
   ) {}
 
-  convertValue(value: ProtoValue): unknown {
+  convertValue(
+    value: ProtoValue,
+    serverTimestampBehavior: ServerTimestampBehavior = 'none'
```

Comment:
Why can't this remain a constructor parameter? That seemed like a cleaner arrangement.

Reply:
Since I am now "injecting" the UserDataWriter when the DocumentSnapshot is constructed, I don't yet know what the setting for `serverTimestampBehavior` will be.